spec/amp-framework-hosting.md

```diff
@@ -0,0 +1,234 @@
+
+The API must meet the following requirements:
+
+- Satisfy [CORS security in AMP](https://github.com/ampproject/amphtml/blob/master/spec/amp-cors-requests.md)
+- Be secure (HTTPS)
+- Return `application/json` content conforming to the following schema:
+  ```
+  {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+      "country": {
+        "type": "string",
+        "title": "ISO 3166-1 alpha-2 (case insensitive) country code of client request",
+        "default": "",
+        "pattern": "^[a-zA-Z]{2}$"
+      }
+    },
+    "required": [
+      "country"
+    ]
+  }
+  ```
+
+A sample response for a user in Germany looks like:
+
+```
+{
+  "country": "de"
+}
+```
+
+There are trade-offs in accuracy and performance when you set the client cache time for this API. Longer cache times are better for performance on subsequent page loads but can lead to incorrect country detection. For reference, `https://cdn.ampproject.org/v0/amp-geo-0.1.js` has a client cache time of 30 minutes.
+
+### amp-optimizer
+
+[amp-toolbox](https://github.com/ampproject/amp-toolbox) has tools ready to help you get the most out of your hosted AMP framework. Notably, the [amp-optimizer](https://github.com/ampproject/amp-toolbox/tree/master/packages/optimizer) tool supports transformation options to use your own host with versioned or versionless URLs in AMP pages. If you're aiming to get the best performance possible out of your AMP pages, this is a great tool to accomplish the task.
+
+### HTTP response Headers
+
+In addition to following [TLS best practices](https://infosec.mozilla.org/guidelines/web_security), be sure to consider the following headers when hosting the AMP framework:
+
+- `content-security-policy`: If your pages implement [AMP's CSP](https://amp.dev/documentation/guides-and-tutorials/optimize-and-measure/secure-pages/), apply a matching content security policy to your hosted framework responses. Inspect the headers on `https://cdn.ampproject.org/v0.js` for a base policy that should be expanded to include resources served from your host.
+- `access-control-allow-origin`: Some runtime components are fetched via XHR. If your AMP pages will be served from a different host than your framework, be sure to include CORS headers (see also [CORS Requests in AMP](https://github.com/ampproject/amphtml/blob/master/spec/amp-cors-requests.md)).
+- `content-type`: There are a few resources served without file extensions, or with extensions that may not be recognized by all web servers. In addition to [common types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types), you may want to include special handling for the following:
+
+  - `/rtv/metadata` - `application/json`
+  - `/amp_preconnect_polyfill_404_or_other_error_expected._Do_not_worry_about_it` - `text/html`
```

Comment:
@rcebulko helped list files that are hosted in each RTV release at cdn.ampproject.org. See #25989. I hope to enhance this guide in the future with a description of these additional files for publishers that opt to build the framework themselves.

Reply:
@ithinkihaveacat I'm not sure why exactly it's here, but it's been a part of our release script for a long time (would have to dig through history to find out why). It can probably be removed at some point, and that may happen during a fixit week or something, but `files.txt` was updated to make sure it contains the exact list of files being served (even if they have no consumers).